docs/ADR-56-avatar-assembling-for-visualization.md

```diff
@@ -0,0 +1,161 @@
+
+```json
+{
+  "bodyShapes":[
+     "urn:decentraland:off-chain:base-avatars:BaseMale"
+  ],
+  "mainFile":"F_Eyewear_BlackSunglasses.glb",
+  "overrideReplaces":[],
+  "overrideHides":[],
+  "contents":[
+     {
+        "key":"AvatarWearables_TX.png",
+        "hash":"QmRaHnacT5G7oLYTYGsRWZtLXzXuTNEq7gWAcvXRSxfwEU"
+     },
+     {
+        "key":"F_Eyewear_BlackSunglasses.glb",
+        "hash":"QmXLrokLeJkmyFD64xGicQyD3R3M4AWcPFLXJMNQx9U7eK"
+     }
+  ]
+}
+```
+
+# Required Categories
+
+Avatars shouldn’t run naked around the world, to prevent this a fallback system was designed when something goes wrong. Only missing wearables in specific categories will be fallen back.
+
+- *BaseFemale*
+    
+    ```
+    "eyes" => "urn:decentraland:off-chain:base-avatars:f_eyes_00"
+    "eyebrows" => "urn:decentraland:off-chain:base-avatars:f_eyebrows_00"
+    "mouth" => "urn:decentraland:off-chain:base-avatars:f_mouth_00"
+    "hair" => "urn:decentraland:off-chain:base-avatars:standard_hair"
+    "upper_body" => "urn:decentraland:off-chain:base-avatars:f_sweater"
+    "lower_body" => "urn:decentraland:off-chain:base-avatars:f_jeans"
+    "feet" => "urn:decentraland:off-chain:base-avatars:bun_shoes"
+    ```
+    
+- *BaseMale*
+    
+    ```
+    "eyes" => "urn:decentraland:off-chain:base-avatars:eyes_00"
+    "eyebrows" => "urn:decentraland:off-chain:base-avatars:eyebrows_00"
+    "mouth" => "urn:decentraland:off-chain:base-avatars:mouth_00"
+    "hair" => "urn:decentraland:off-chain:base-avatars:casual_hair_01"
+    "facial" => "urn:decentraland:off-chain:base-avatars:beard"
+    "upper_body" => "urn:decentraland:off-chain:base-avatars:green_hoodie"
+    "lower_body" => "urn:decentraland:off-chain:base-avatars:brown_pants"
+    "feet" => "urn:decentraland:off-chain:base-avatars:sneakers"
+    ```
+    
+
+# Cheat sheet
+
+This is a summary of the process to render an avatar.
+
+1. Gather the *bodyshape.*
+2. Filter out *wearables* that doesn’t contain a *representation* for that *bodyshape.*
+3. Iterate the hides list and remove affected *wearables* from the process.
+4. Fill *required categories* with default *wearables* for the *bodyshape.*
+5. Download the wearable assets.
+    1. If the download fails, fallback to a default one.
+6. Apply coloring to hair and skin.
+
+# Armature
+
+Each wearable carries its own set of bones (only the needed ones) which makes them completely autonomous and capable of playing animations on their own.
+
+# Shading
+
+Three different shaders are used when rendering an avatar.
+
+- *Unlit Cutout with Tint*: Used for mouth and eyebrows: The tint can be masked with a monochannel texture, *black* tints*, white* doesnt tint.
+- *Unlit Cutout for Eyes*: It’s similar to the previous one but this one contains a mask where the iris is tinted.
+- *Toon shader:* The rest of the avatar is rendered with a *Toon Shader.*
```

Comment:
Do the wearables use toon shader or other material?

Reply:
Yes, anything that is not mouth/eyes/eyebrows uses the same toon shader. I didn't go into details about toon shader because it's really tight to Unity, HLSL and its render pipeline. In fact the decision of using a toon shader is sorely artistic, nothing in the wereables definitions prevent you from implementing any kind of lightning interpretation you want.

Wearables do not carry PBR textures, but nothing should stop somebody from having an Avatar visualization with a photorealistic approach (ala Minecraft with ray-tracing).